File: tests/human/x/zig/cross_join_triple.zig
const std = @import("std");

const Combo = struct {
    n: i32,
    l: []const u8,
    b: bool,
};

pub fn main() !void {
    const nums = [_]i32{1, 2};
    const letters = [_][]const u8{"A", "B"};
    const bools = [_]bool{ true, false };

    var combos = std.ArrayList(Combo).init(std.heap.page_allocator);
    defer combos.deinit();

    for (nums) |n| {
        for (letters) |l| {
            for (bools) |b| {
                try combos.append(.{ .n = n, .l = l, .b = b });
            }
        }
    }

    std.debug.print("--- Cross Join of three lists ---\n", .{});
    for (combos.items) |c| {
        std.debug.print("{d} {s} {}\n", .{ c.n, c.l, c.b });
    }
}

File: tests/human/x/python/cross_join_triple.py
nums = [1, 2]
letters = ["A", "B"]
bools = [True, False]
combos = [
    {"n": n, "l": l, "b": b}
    for n in nums
    for l in letters
    for b in bools
]
print("--- Cross Join of three lists ---")
for c in combos:
    print(c["n"], c["l"], c["b"])
